addons/delivery/models/sale_order.py

```diff
@@ -50,12 +50,17 @@ def delivery_set(self):
         for order in self:
             carrier = order.carrier_id
             if carrier:
+                service_charges = None
                 if order.state not in ('draft', 'sent'):
                     raise UserError(_('The order state have to be draft to add delivery lines.'))
 
                 if carrier.delivery_type not in ['fixed', 'base_on_rule']:
                     # Shipping providers are used when delivery_type is other than 'fixed' or 'base_on_rule'
-                    price_unit = order.carrier_id.get_shipping_price_from_so(order)[0]
+                    result = order.carrier_id.get_shipping_price_from_so(order)
+                    price_unit = result[0]
+                    if len(result) > 1:
+                        price_unit = result[1]
+                        service_charges = result[2]
```

Comment:
Hacky hack, please explain in a comment why result can be off different lenghts.